### DISCUSSION TITLE
Using Quest Passthrough with MRTK3

### DISCUSSION THREAD

**Discussion by thisisvaze** [microsoft/MixedRealityToolkit-Unity#11239] [Q&A] [answered]
How to develop apps with passthrough on a Quest device using MRTK3?

**Comment by thisisvaze** [comment, marked as answer]
I was able to use the passthrough mode on a Quest device as suggested by @Zee2. 
1. Add both MRTK3 and Oculus Integration SDK to your Unity project.
2. In the scene use MRTK XR Rig and add these scripts: OVR Manager and OVR Passthrough layer.
3. The only issue was that the MRTK XR Rig's Main camera sets the background black when initiated, so I added this to Update() on a script with the following code to change the background color to clear.
`Camera.main.backgroundColor = new Color(0f,0f,0f,0f);`

Tested and worked in the Unity play mode, not sure if it works when the app is deployed.
Can't wait to get official support for the passthrough but this should do for prototyping until then!

**Reply by trzy** [reply]
This is what I've been doing but I can't get controllers to show up and the hand constraints for UI objects (e.g., to create a hand-attached menu) don't work at all. Have you been able to get these working?

**Reply by trzy** [reply]
To be clear, the controller nodes seem to be partially working. For example, the rays are emitted correctly and I can interact with buttons. But the HandConstraint script doesn't seem to be able to find any position to attach objects to.

**Reply by Zee2** [reply]
Hey @trzy, known issue with the current handconstraint script. Refactor to a standalone handmenu script is happening here: #10852

**Reply by Zee2** [reply]
(current handconstraint script has no fallback when no joints are available.)

**Reply by keveleigh** [reply]
> The only issue was that the MRTK XR Rig's Main camera sets the background black when initiated

Huh, I can't seem to find the component that's doing that and have been unable to repro...we have a script called `CameraSettingsManager` which allows for specifying your camera settings across transparent and opaque devices, but it's not on the MRTK XR Rig prefab by default and can also be customized in the inspector. The rig's camera appears to be `Color.clear` by default. Please let me know if I'm missing anything so I can help streamline this path!

**Reply by thisisvaze** [reply]
@keveleigh You can add this script Camera.main.backgroundColor = new Color(0f,0f,0f,0f); in your script on Update() method and it should override the previous default settings.

**Reply by LocalJoost** [reply]
@thisisvaze this does not seem to work when deployed. Would you mind sharing your OVR Passthrough Layer script setting? If I set it to Overlay it kinda works, but I have to set the opacity to 0.3 so it's all kind of hazy. If I set it to Underlay I basically don't see any background.
